packages/app-desktop/gui/DefaultEditorDialog/Dialog.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import { useCallback } from 'react';
+import { _ } from '@joplin/lib/locale';
+import DialogButtonRow from '../DialogButtonRow';
+import Dialog from '../Dialog';
+import DialogTitle from '../DialogTitle';
+import Button, { ButtonLevel } from '../Button/Button';
+import bridge from '../../services/bridge';
+import Setting from '@joplin/lib/models/Setting';
+
+interface Props {
+	themeId: number;
+	onClose: ()=> void;
+}
+
+interface EditorInfo {
+	name: EditorTargetInfoName;
+	description: string;
+}
+
+const editorOptions: string[] = [
+	'Rich Text Editor',
+	'Markdown Editor',
+];
+
+
+const logosImageNames: Record<string, string> = {
+	'Rich Text Editor': 'default-editor_RTE_icon.png',
+	'Markdown Editor': 'default-editor_Markdown_icon.png',
+};
+
+const editorExamples: Record<string, string> = {
+	'Rich Text Editor': 'default-editor_RTE.png',
+	'Markdown Editor': 'default-editor_Markdown.png',
+};
+
+enum EditorTargetInfoName { RTE='Rich Text Editor' , Markdown='Markdown Editor'};
```

Comment:
```suggestion
enum EditorTargetInfoName { RTE='Rich Text Editor' , Markdown='Markdown Editor'};

const logosImageNames: Record<string, string> = {
	[EditorTargetInfoName.RTE]: 'default-editor_RTE_icon.png',
	[EditorTargetInfoName.Markdown]: 'default-editor_Markdown_icon.png',
};

const editorExamples: Record<string, string> = {
	[EditorTargetInfoName.RTE]: 'default-editor_RTE.png',
	[EditorTargetInfoName.Markdown]: 'default-editor_Markdown.png',
};
```

**(structure/type safety)**:

- Please use the `enum` where possible. This should allow for additional type checking.
- It looks like the values are the same for both. Can these be combined into a single object?

Reply:
**(idea/code structure)**: Additionally, it could make sense to pass this information as props to subcomponents rather than store this information in `Record`s. This might be done, for example, by creating a new `EditorChoice` component. This component could be used in a way similar to the following:
```tsx
   <div className="editorTargetBoxes">
      <EditorChoice
          icon='default-editor_RTE_icon.png'
          logo={`${logoDir}/RTE.png`}
          settingKey='RTE'
          title={_('Rich Text Editor')}
          description={_('Similar to Word and other WYSIWYG editors')}
          onAfterSelected={props.onClose} />
      <EditorChoice
          ...similar for the markdown editor... />
     </div>
```

Doing this would keep the icon/description/settings key information for each editor in one place and could make this information easier to update in the future.